### ISSUE TITLE
Pull kernel functions out of operator classes

### ISSUE THREAD

**Issue by lockshaw** [flexflow/flexflow-train#303]
Following https://github.com/lockshaw/FlexFlow/blob/feat/candle_split/include/flexflow/ops/linear_kernels.h

See discussion at https://github.com/flexflow/FlexFlow/pull/298#discussion_r956470999

**Comment by lockshaw** [comment]
A more up-to-date example is shown in #320

**Comment by lockshaw** [comment]
List of operators: 

- `BatchMatmul`
- `Conv2D` (#434)
- `Concat`
- `Cast`
- `ElementBinary`
- `ElementUnary`
- `Dropout`
- `Embedding`
- `Flat`
- `LayerNorm`
- `Linear`
- `MultiHeadAttention`
- `NoOp/Input`
- `Pool2D`
- `Reshape`
- `Split`
- `Softmax`
- `Transpose`
- `Repartition`
- `Replicate`
- `Reduction`
- `Combine`
- `FusedParallelOp`

**Comment by lockshaw** [comment]
Closed by #622.
